### ISSUE TITLE
[i2c] Improve the `I2cAdapterTrait`

### ISSUE THREAD

**Issue by andreeaflorescu** [rust-vmm/vhost-device#26]
There are a few odd things in the trait. For example, get methods that need a mutable reference, and set methods that DO NOT need a mutable reference:

```rust
    /// Sets device's address for an I2C adapter.
    fn set_device_addr(&self, addr: usize) -> Result<()>;

    /// Gets adapter's functionality
    fn get_func(&mut self) -> Result<()>;
```

Is this a naming problem or something else?

**Comment by andreeaflorescu** [comment]
Also, the `get_func` returns an empty result.

**Comment by vireshk** [comment]
set_device_addr() sets the address at the i2c adapter at the host, and so mut isn't required.

I think we can get rid of get_func() and merge it with new() itself

**Comment by andreeaflorescu** [comment]
> set_device_addr() sets the address at the i2c adapter at the host, and so mut isn't required.
> 
> I think we can get rid of get_func() and merge it with new() itself

Yes, that'd be my suggestion as well. This also helps with making the function `is_smbus` always return a valid answer irrespective of when it is called.
